### PR TITLE
BIDS import - handle multiple physiology files and derivatives

### PR DESCRIPTION
**Pull request by laemtl**
The current version of eeg.py handles the import of a single physiology file per subject/modality folder. This extends the current code to import all the physiology files present with their appropriate metadata.

-- note: 
~~The derivative case is broken until I revisit #608 ([code L162-165](https://github.com/aces/Loris-MRI/pull/608/files#diff-250dc8b14f8948bfc4cf003892b0da4257fc45681b3e49d5a7af615e552ea5bbR163) is already commented so it does not break anything).~~
The derivative case is fixed and needs testing.

### CODE REVIEW

**Review by cmadjar — CHANGES_REQUESTED** [review]
@laemtl This is working well for the Face13 dataset that is on raisinbread as well as on fake datasets with multiple runs of EDF files.

When testing some of the bids-examples, I encountered two issues:
- An issue for support of SET and FDT files when testing https://github.com/bids-standard/bids-examples/tree/master/eeg_ds000117. See comment on code for line 401. 
- the inheritance of JSON file when the file is not in the eeg folder is not working. See comment on code for line 407.

**Comment by cmadjar** [comment]
@laemtl final case test I did not see before:

- when there are two generic electrodes file in the eeg folder. For example:
```
sub-02_coordsystem.json
sub-02_task-faceO_electrodes.tsv
sub-02_task-faceO_run-2_eeg.edf
sub-02_task-faceO_run-2_events.tsv
sub-02_task-facerecognition_electrodes.tsv
sub-02_task-facerecognition_run-1_eeg.edf
sub-02_task-facerecognition_run-1_events.tsv
```

This one is a silent failure. Instead of inserting `task-facerecognition_electrodes.tsv` for the `task-facerecognition_run-1_eeg.edf`, the pipeline is inserting the file and data for `sub-02_task-faceO_electrodes.tsv`. Same thing for the events file. So the resulting structure inserted into LORIS is:
```
sub-02_ses-V1_task-faceO_electrodes.tsv
sub-02_ses-V1_task-faceO_run-2_eeg.edf
sub-02_ses-V1_task-faceO_run-2_eeg.tgz
sub-02_ses-V1_task-faceO_run-2_events.tsv
sub-02_ses-V1_task-facerecognition_run-1_eeg.edf
sub-02_ses-V1_task-facerecognition_run-1_eeg.tgz
```

Note: looks like the coordsystem.json file is never inserted either, which is a different issue than this PR.

**Comment by cmadjar** [comment]
@laemtl I am adding the Needs work before merging label so I stop looking into that PR. Once you are done working on it, you can remove the label and tag me so I can re-review. Hope that works. Thanks!

**Comment by laemtl** [comment]
@cmadjar Ready for second review. I addressed the derivatives case so I updated the PR title accordingly.

**Comment by cmadjar** [comment]
@laemtl just FYI, there were conflicts with the main branch that I just resolved.

File: python/lib/eeg.py
Comment (cmadjar) [inline python/lib/eeg.py:315]: why are those two commented?

File: python/lib/bidsreader.py
Comment (cmadjar) [inline python/lib/bidsreader.py:107]: This does not work for datasets that have no derivatives. I get the following warnings trying to load the Face13 dataset with no derivatives:
```
/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/bids/layout/validation.py:131: UserWarning: Derivative indexing was requested, but no valid datasets were found in the specified locations (['/Users/cmadjar/Data/LORIS_imaging_data/incoming/EEG/Face13_small/derivatives']). Note that all BIDS-Derivatives datasets must meet all the requirements for BIDS-Raw datasets (a common problem is to fail to include a 'dataset_description.json' file in derivatives datasets).
Example contents of 'dataset_description.json':
{"Name": "Example dataset", "BIDSVersion": "1.0.2", "PipelineDescription": {"Name": "Example pipeline"}}
  warnings.warn("Derivative indexing was requested, but no valid "
```

File: python/lib/physiological.py
Comment (cmadjar) [inline python/lib/physiological.py:544]: this is causing an error for datasets that do not have an event_value column in the events file (example: Face13 dataset that has only onset, duration and trial_type set in the TSV file). 

So in this case, row['event_value'] and row['value'] are set to `None` and when you convert it to `str` it tries to insert the string "None" in the `EventValue` field that should be a 10 digits integer causing the following failure:

```
Traceback (most recent call last):
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/database.py", line 185, in insert
    cursor.executemany(query, values)
  File "/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 230, in executemany
    return self._do_execute_many(
  File "/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 261, in _do_execute_many
    rows += self.execute(sql + postfix)
  File "/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 206, in execute
    res = self._query(query)
  File "/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/MySQLdb/cursors.py", line 319, in _query
    db.query(q)
  File "/Users/cmadjar/Development/python_venv/python_intel_3.9.4/loris-mri-venv/lib/python3.9/site-packages/MySQLdb/connections.py", line 259, in query
    _mysql.connection.query(self, query)
MySQLdb._exceptions.OperationalError: (1366, "Incorrect integer value: 'None' for column `loris_rb`.`physiological_task_event`.`EventValue` at row 1")

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/cmadjar/Development/Github/loris-mri/python/bids_import.py", line 463, in <module>
    main()
  File "/Users/cmadjar/Development/Github/loris-mri/python/bids_import.py", line 76, in main
    read_and_insert_bids(bids_dir, config_file, verbose, createcand, createvisit)
  File "/Users/cmadjar/Development/Github/loris-mri/python/bids_import.py", line 220, in read_and_insert_bids
    Eeg(
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/eeg.py", line 153, in __init__
    self.register_data()
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/eeg.py", line 260, in register_data
    event_file_path     = self.fetch_and_insert_event_file(eeg_file_id, original_file_data.path, derivatives)
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/eeg.py", line 637, in fetch_and_insert_event_file
    physiological.insert_event_file(
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/physiological.py", line 551, in insert_event_file
    self.db.insert(
  File "/Users/cmadjar/Development/Github/loris-mri/python/lib/database.py", line 193, in insert
    raise Exception("Insert query failure: " + format(err))
Exception: Insert query failure: (1366, "Incorrect integer value: 'None' for column `loris_rb`.`physiological_task_event`.`EventValue` at row 1")
```

**Review by cmadjar — CHANGES_REQUESTED** [review]
@laemtl I did an initial test with the Face13 dataset from James Desjardins that is on demo and noticed the following issues.

I also tested on a dataset that I created with multiple EEG files and it worked very nicely :).

File: python/lib/eeg.py
Comment (laemtl) [reply python/lib/eeg.py:315]: Those line mimic the previous logic, but are commented because I think it may be good to not be too restrictive. You can let me know your though and I will adjust accordingly.

File: python/lib/bidsreader.py
Comment (laemtl) [reply python/lib/bidsreader.py:107]: Weird, I think I tested the same case without any problem. Let me double-check.

Comment (laemtl) [reply python/lib/bidsreader.py:107]: Ok, I see, it's a warning, not an error. Does it prevent the pipeline to run?

File: python/lib/physiological.py
Comment (laemtl) [reply python/lib/physiological.py:544]: Good catch, it's easy to solve.

Comment (laemtl) [reply python/lib/physiological.py]: @cmadjar Fixed.

File: python/lib/bidsreader.py
Comment (cmadjar) [reply python/lib/bidsreader.py:107]: All good, we can keep the warning. I initially thought that was why the pipeline failed when I tested it but it was because of the task event insertion.

File: python/lib/eeg.py
Comment (cmadjar) [reply python/lib/eeg.py:315]: I would tend to keep them to ensure you are getting only the EEG or iEEG files. For example, maybe for MEG or other electrophysiological recordings, the files will have the same extensions as EEG but their insertion process might be different than EEG? 
At least, we are sure we will not get the MRIs since the extension of images are `.nii.gz` but relying only on file extensions to grep the proper files seem risky. What do you think?

Comment (laemtl) [reply python/lib/eeg.py:315]: Given your reasoning, I agree with you. Fixed!

**Review by cmadjar — APPROVED** [review]

**Pull request merged**
